shared_model/interfaces/queries/responses/transactions_response.hpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_TRANSACTIONS_RESPONSE_HPP
+#define IROHA_SHARED_MODEL_TRANSACTIONS_RESPONSE_HPP
+
+#include <boost/algorithm/string/join.hpp>
+#include <rxcpp/operators/rx-sequence_equal.hpp>
+#include <rxcpp/rx-observable.hpp>
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/hashable.hpp"
+#include "interfaces/queries/responses/transactions_response.hpp"
+#include "interfaces/transaction.hpp"
+#include "model/queries/responses/transactions_response.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Container of asset, for fetching data.
+     */
+    class TransactionsResponse
+        : public Hashable<TransactionsResponse,
+                          iroha::model::TransactionsResponse> {
+     public:
+      /**
+       * @return Attached transactions
+       */
+      // TODO 01/11/17 motxx - Change detail::PolymorphicWrapper ->
+      // PolymorphicWrapper or use concrete type for Transaction
+      virtual rxcpp::observable<detail::PolymorphicWrapper<Transaction>>
+      transactions() const = 0;
+
+      /**
+       * Stringify the data.
+       * @return string representation of data.
+       */
+      virtual std::string toString() const override {
+        std::vector<std::string> vs;
+        transactions().subscribe(
```

Comment:
I'm not that profound in rx, but I don't really sure that it will work.
Don't the `subscribe` unblocking call? ping @muratovv

Reply:
All subscribe calls are blocking. For performing unblocking require to use `onObservable()` that will perform subscriber call on another thread.

Reply:
I see, that's fine then